packages/near-membrane-base/src/membrane.ts

```diff
@@ -725,7 +768,9 @@ export function createMembraneMarshall() {
                 if (foreignTargetTraits & TargetTraits.Revoked) {
                     revoke();
                 }
-                if (!trapMutations) {
+                if (foreignTargetTraits & TargetTraits.IsArray) {
+                    this.makeProxyLive();
```

Comment:
But this should only really happen if `trapMutations` is true, otherwise for blue this line might get called when it is already the case that the default hooks are live (in the case of the blue side for example)

Reply:
You're right. Blue is different than live.

Reply:
Fixed.